### DISCUSSION TITLE
How can I get the Facebook User ID from my users?

### DISCUSSION THREAD

**Discussion by diamondback15** [strapi/strapi#7402] [Legacy Topics]
I'm migrating a large db of users from Strapi to Firebase and I need to provide the providerData.uid coming from the Facebook accounts. Is there a way to get theses id from Strapi? Does Strapi save the IDs somewhere?

**Comment by derrickmehaffy** [comment]
Hi @diamondback15 we don't currently query for the user ID, here is the related code where we query the Facebook API for data: https://github.com/strapi/strapi/blob/e25a508b2bc090546186c7ea1b2f84a4b5b5c428/packages/strapi-plugin-users-permissions/services/Providers.js#L164-L185

You will want to use the [extensions](https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#how-do-i-customize-a-plugin) system to override this file if you need additional data

**Reply by diamondback15** [reply]
Ok thank you. So Strapi is matching the provider email with the user email everytime someone login

**Reply by derrickmehaffy** [reply]
you can extend that file and ask for more information (depending on what facebook allows, to that I'm not sure) and you can also assign that extra data to fields within the user model should you have custom fields.
